### PR TITLE
Use specific errors for intent token revoked

### LINKED ISSUE #3288: Unhanded error when using credential reset token
### I did this
Used a reset token and attempted to save TOTP & password. Got InvalidState. After clearing cookies TOTP and password saving was possible.

### I expected the following
TOTP & password to be saved successfully or an error message informing me to restart the flow due to timeout.

### This happened instead
InvalidState returned.
Logs:
```
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 INFO     request [ 15.5ms | 92.70% / 100.00% ] method: POST | uri: /ui/api/cu_commit | version: HTTP/1.1
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 INFO     ┝━ handle_idmcredentialupdatecommit [ 1.13ms | 7.30% ]
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 INFO     │  ┝━ ｉ [info]: Session originated from an intent token, but the intent token has initiated a conflicting second update session. Refusing to commit changes. | event_tag_id: 10
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 ERROR    │  ┕━ 🚨 [error]: Failed to begin commit_credential_update | err: InvalidState
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 ERROR    ┝━ 🚨 [error]:  | latency: 1.280185ms | status_code: 500 | kopid: "723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7" | msg: "error handling request"
723c8bb2-5c3f-45d8-b0a7-1e9f9d7f2bc7 ERROR    ┕━ 🚨 [error]: response failed | classification: Status code: 500 Internal Server Error | latency: 1 ms
```
### Kanidm version details

- Output of `kanidm(d) version`: `kanidmd 1.4.4`
- Are you running it in a container? If so, which image/tag?: `1.4.4`
- If not a container, how'd you install it:
- Operating System / Version (On Unix please post the output of `uname -a`): `6.1.0-28-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.119-1 (2024-11-22) x86_64 GNU/Linux`

### PR DESCRIPTION
**Pull request by Firstyear**
Rather than the generic 'invalid state' error, we now return proper site-specific errors for credential commit failures, with error messages to explain what went wrong.

Fixes #3288

Checklist

- [x] This PR contains no AI generated code
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: server/lib/src/idm/credupdatesession.rs [hunk 1]
```diff
@@ -1312,7 +1312,7 @@ impl IdmServerProxyWriteTransaction<'_> {
                 commit_failure_reasons
             );
             // TODO: perhaps it would be more helpful to add a new operation error that describes what the issue is
```
Comment (gbw) [inline server/lib/src/idm/credupdatesession.rs:1314]: Remove TODO?

**Review by CEbbinghaus — DISMISSED** [review]
No additional comments beyond the todo

File: proto/src/internal/error.rs [hunk 2]
```diff
@@ -364,6 +373,11 @@ impl OperationError {
             Self::CU0001WebauthnAttestationNotTrusted => None,
             Self::CU0002WebauthnRegistrationError => None,
             Self::CU0003WebauthnUserNotVerified => Some("User Verification bit not set while registering credential, you may need to configure a PIN on this device.".into()),
+
+            Self::CU0004SessionInconsistent => Some("The session is unable to be committed due to unresolved warnings".into()),
```
Comment (yaleman) [inline proto/src/internal/error.rs:377]: All sentences need to end with appropriate punctuation

**Review by yaleman — CHANGES_REQUESTED** [review]
The security_info log calls at the end could probably be dropped because we have an informative error code but otherwise looks good so far 😊

**Comment by Firstyear** [comment]
Applied review feedback.

**Review by gbw — APPROVED** [review]
Good after rebase.

**Review by yaleman — APPROVED** [review]
:uwu:

**Pull request merged**